tools/internal_ci/linux/grpc_run_tests_matrix.sh

```diff
@@ -22,6 +22,7 @@ source tools/internal_ci/helper_scripts/prepare_build_linux_rc
 
 # If this is a PR using RUN_TESTS_FLAGS var, then add flags to filter tests
 if [ -n "$KOKORO_GITHUB_PULL_REQUEST_NUMBER" ] && [ -n "$RUN_TESTS_FLAGS" ]; then
+  sudo apt-get update
```

Comment:
nit: installing this here is pretty dirty, ideally we would do this in a cleaner way. Let's keep this in mind when addressing technical debt.